### PR TITLE
Bluetooth: Mesh: splitting between active and define states for scheduled register

### PR DESCRIPTION
**Pull request by alxelax**
After clarification in Bluetooth model specification,
the defined scheduled register is an entry where the action is not equal
to 'Inactive'.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/mesh/scheduler_srv.c [hunk 1]
```diff
@@ -730,11 +746,17 @@ static void scheduler_srv_reset(struct bt_mesh_model *model)
...
+	srv->active_bitmap = 0;
 	k_delayed_work_cancel(&srv->delayed_work);
 	net_buf_simple_reset(srv->pub.msg);
 	memset(&srv->sch_reg, 0, sizeof(srv->sch_reg));
 
+	for (int i = 0; i < BT_MESH_SCHEDULER_ACTION_ENTRY_COUNT; i++) {
+		srv->sched_tai[i].sec = 0;
+		srv->sched_tai[i].subsec = 0;
+		srv->sch_reg[i].action = BT_MESH_SCHEDULER_NO_ACTIONS;
+	}
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/scheduler_srv.c:758]: Is both this and the memset necessary? Couldn't we just set `.action` for all, as that's the decision point?

**Review by trond-snekvik — COMMENTED** [review]
It'd be nice to wrap the settings and comparison of `.action` in `is_defined` and `clear` like functions, just to convey that this parameter is the decision point

File: subsys/bluetooth/mesh/scheduler_srv.c (hunk 1 above)
Comment (alxelax) [reply subsys/bluetooth/mesh/scheduler_srv.c:758]: I guess it is possible to not set zeroes in tai structure. The server will have garbage there until it is configured. Not nice from my point of view but does not impact anything.
However, I'm not sure it is a good idea to store garbage into flash if memset is removed.

**Comment by alxelax** [comment]
> It'd be nice to wrap the settings and comparison of `.action` in `is_defined` and `clear` like functions, just to convey that this parameter is the decision point

I didn't get about `clear`. `action` has more than `defined` and `not defined` states. Table 5.12: Action field values

File: subsys/bluetooth/mesh/scheduler_srv.c (hunk 1 above)
Comment (trond-snekvik) [reply subsys/bluetooth/mesh/scheduler_srv.c:758]: Could we just delete the unused entires from flash instead?

**Comment by trond-snekvik** [comment]
> I didn't get about `clear`. `action` has more than `defined` and `not defined` states. Table 5.12: Action field values

Yes, but the code would be clearer if instead of 

```c
if (srv->sch_reg[i] != BT_MESH_SCHEDULER_NO_ACTIONS) {
    // do something
}
```
we had
```c
if (is_defined(&srv->sch_reg[i])) {
    // do something
}
```

The underlying parameter and its value can stay the same, but the intent behind the NO_ACTIONS check is a bit hard to follow

File: subsys/bluetooth/mesh/scheduler_srv.c (hunk 1 above)
Comment (omkar3141) [reply subsys/bluetooth/mesh/scheduler_srv.c:758]: Deletion will add yet another write to the entry and entry name and will make flash wear worse (name + value both are removed). It does not buy anything. Even if stack deletes entries now at some point they will get used anyway.

File: subsys/bluetooth/mesh/scheduler_srv.c [hunk 2]
```diff
@@ -554,6 +568,8 @@ static int send_scheduler_action_status(struct bt_mesh_model *model,
 	struct bt_mesh_scheduler_srv *srv = model->user_data;
 
 	BT_MESH_MODEL_BUF_DEFINE(buf, BT_MESH_SCHEDULER_OP_ACTION_STATUS,
+		srv->sch_reg[idx].action == BT_MESH_SCHEDULER_NO_ACTIONS ?
```
Comment (omkar3141) [inline subsys/bluetooth/mesh/scheduler_srv.c:571]: This is not fully correct. When Scheduler Action Status is sent in response to Scheduler Action Set, the Scheduler Register field should be the same as a request message.

**Review by omkar3141 — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/mesh/scheduler_srv.c (hunk 1 above)
Comment (alxelax) [reply subsys/bluetooth/mesh/scheduler_srv.c:767]: deleted both in init and reset

**Review by trond-snekvik — APPROVED** [review]

**Review by omkar3141 — APPROVED** [review]
